### PR TITLE
balance tx with additional utxo map

### LINKED ISSUE #1000: Transaction evaluation with additional UTXO set
I'm trying to get depending transactions into the mempool.
This should work when submitting the transactions in the proper order, but before we can submit them, we have to evaluate them.
To evaluate a transaction with UTXOs that do not exist yet, Ogmios provides the additionalUtxoSet option where we can submit non-existing inputs.
As I can see in the code, CTL does not support such evaluation yet.

This issue is related: https://github.com/Plutonomicon/cardano-transaction-lib/issues/310

### PR DESCRIPTION
**Pull request by the-headless-ghost**
Closes #1000 .

@klntsky Can you please check if this is according to the guidelines and what else is needed? I did a rough test. I successfully chained three dependent transactions into one block.

### Pre-review checklist

- [X] All code has been formatted using our config (`make format`)
- [ ] Any new API features or modification of existing behavior is covered as defined in our [test plan](https://github.com/Plutonomicon/cardano-transaction-lib/blob/develop/doc/test-plan.md)
- [ ] The changelog has been updated under the `## Unreleased` header, using the appropriate sub-headings (`### Added`, `### Removed`, `### Fixed`), and the links to the appropriate issues/PRs have been included

### CODE REVIEW

**Comment by klntsky** [comment]
@the-headless-ghost sorry for slow response. Could you please resolve the conflicts?

File: src/BalanceTx/ExUnitsAndMinFee.purs [hunk 1]
```diff
@@ -8,6 +8,20 @@ import Prelude
 import BalanceTx.Error
   ( BalanceTxError(ExUnitsEvaluationFailed, ReindexRedeemersError)
   )
+-- <<<<<<< HEAD
```
Comment (klntsky) [inline src/BalanceTx/ExUnitsAndMinFee.purs:11]: Please remove this leftover

**Review by klntsky — CHANGES_REQUESTED** [review]

File: src/BalanceTx/ExUnitsAndMinFee.purs (hunk 1 above)
Comment (the-headless-ghost) [reply src/BalanceTx/ExUnitsAndMinFee.purs:11]: oh right, totally forgot.

**Comment by the-headless-ghost** [comment]
I just noticed that the `UtxoMap` type changed. Instead of `TransactionOutput` we have `TransactionOutputWithRefScript`. I don't know if this still makes sense in this context. Should I keep this type, or should I change it to `Map TransactionInput TransactionOutput`?

**Comment by klntsky** [comment]
Reopening as #1046. We need it to be on a local branch for CI to run.

**Pull request closed without merging**